frappe/public/js/frappe/form/form.js

```diff
@@ -1144,6 +1144,10 @@ frappe.ui.form.Form = class FrappeForm {
 						me.refresh_field(f);
 						return Promise.resolve();
 					} else {
+						if (fieldobj.parse) {
```

Comment:
This should be fixed in Barcode field, not here.

Reply:
Done